packages/table/src/tableBody.tsx

```diff
@@ -5,24 +5,30 @@
  * and https://github.com/palantir/blueprint/blob/master/PATENTS
  */
 
-import { ContextMenuTarget, IProps } from "@blueprintjs/core";
 import * as React from "react";
-import { emptyCellRenderer, ICellProps, ICellRenderer } from "./cell/cell";
+
+import { ContextMenuTarget, IProps } from "@blueprintjs/core";
+
+import { emptyCellRenderer, ICellProps, ICellRenderer, loadingCellRenderer } from "./cell/cell";
+import { IColumnProps } from "./column";
 import { Grid, IColumnIndices, IRowIndices } from "./common/grid";
+import { ILoadable } from "./common/loading";
 import { Rect } from "./common/rect";
 import { Utils } from "./common/utils";
 import { ICoordinateData } from "./interactions/draggable";
 import { IContextMenuRenderer, MenuContext } from "./interactions/menus";
 import { DragSelectable, ISelectableProps } from "./interactions/selectable";
 import { ILocator } from "./locator";
-import { Regions } from "./regions";
+import { ColumnLoadingOption, Regions } from "./regions";
 
-export interface ITableBodyProps extends ISelectableProps, IRowIndices, IColumnIndices, IProps {
+export interface ITableBodyProps extends IColumnIndices, ILoadable, IProps, IRowIndices, ISelectableProps {
     /**
      * A cell renderer for the cells in the body.
      */
     cellRenderer: ICellRenderer;
 
+    getColumnProps: (columnIndex: number) => IColumnProps;
```

Comment:
**:fire: adding a new required prop is a significant API break.** what is the purpose of this prop? at the very least it's going to need some clear documentation.